proposals/1533-host-specific-tasklist.md

```diff
@@ -0,0 +1,241 @@
+
+The session will be created on one of the workers that poll from the tasklist specified in the context passed into the CreateSession API. As a result, users needs to make sure they are using the correct task list.
+
+When executing an activity within a session, user may get two types of errors. The first one is those returned from user activities. Session will not be marked as failed in this case, so user can return whatever error they want and apply their business logic as necessary. If they want to end a session due to the error returned from the activity, use the CompleteSession() API below. The second type of error is a special error which says the session has failed due to worker failure and the session is marked as failed in the background. In this case, no activities can be executed using this context. It’s the user’s choice on how to handle the failure; for example, they can create a new session to retry or end the workflow with an error.
+
+```go
+err := workflow.**CompleteSession**(sessionCtx)
+```
+
+This API is used to complete a session. It will return an error if the context passed in does not contain a session with state "Open" or it fails to complete the session.
+
+```go
+sessionInfo, err := workflow.**GetSessionInfo**(sessionCtx)
+```
+
+This API returns session information stored in the context. It returns an error if the context passed in doesn’t contain any session information or the session has already failed. For now, the only information returned is sessionID, which is a unique identifier for a session.
+
+```go
+sessionCtx, err := workflow.**RecreateSession**(ctx, sessionInfo)
+```
+
+This API takes session information and recreate a session. All the activities in the recreated session will be executed by the specified worker. The primary use case will be continueAsNew. Before returning the continueAsNew error, user should get the sessionInfo from sessionCtx and complete the session. Then in the next run, the session can be recreated and activities in this new run can continue to be executed on the same activity worker as previous runs.
+
+#### Example
+
+The basic usage looks like the following (it belongs to a larger workflow):
+
+```go
+sessionCtx, err := CreateSession(ctx)
+
+if err != nil {
+
+    // Creation failed. Wrong ctx or too many outstanding sessions.
+
+}
+
+err = ExecuteActivity(sessionCtx, DownloadFileActivity, filename).Get(sessionCtx, nil)
+
+if err != nil {
+
+    // Session(worker) has failed or activity itself returns an error.
+
+    // User can perform normal error handling here and decide whether creating a new session is needed.
+
+    // If they decide to create a new session, they need to call CompleteSession() so that worker resource can be released. It’s recommended that users create a function for session and call defer CompleteSession(sessionCtx) after a session is created. If session retry is needed, this function can be called multiple times if the error returned is retriable.
```

Comment:
so that worker resource -> so that worker resources
It's recommended that -> We recommend that
function for session -> function for a session